tempus_fugit_minecraft/utilities.py

```diff
@@ -0,0 +1,119 @@
+def cube_vertices(x: float, y: float, z: float, n: float) -> list:
+    """Return the vertices of the cube at position x, y, z with size 2*n.
+
+    Parameters
+    ----------
+    x, y, z : float
+        Coordinates of the cube's center.
+    n : float
+        Size of the cube.
+
+    Returns
+    -------
+    vertices : list
+        The vertices of the cube.
+    """
+    return [
+        x-n,y+n,z-n, x-n,y+n,z+n, x+n,y+n,z+n, x+n,y+n,z-n,  # top
+        x-n,y-n,z-n, x+n,y-n,z-n, x+n,y-n,z+n, x-n,y-n,z+n,  # bottom
+        x-n,y-n,z-n, x-n,y-n,z+n, x-n,y+n,z+n, x-n,y+n,z-n,  # left
+        x+n,y-n,z+n, x+n,y-n,z-n, x+n,y+n,z-n, x+n,y+n,z+n,  # right
+        x-n,y-n,z+n, x+n,y-n,z+n, x+n,y+n,z+n, x-n,y+n,z+n,  # front
+        x+n,y-n,z-n, x-n,y-n,z-n, x-n,y+n,z-n, x+n,y+n,z-n,  # back
+    ]
+
+
+def tex_coord(x: int, y: int, n=4) -> tuple:
+    """Return the bounding vertices of the texture square.
+
+    Parameters
+    ----------
+    x, y : int
+        The x, y coordinates of the texture square.
+    n : int
+        The size of the texture atlas (default 4).
+
+    Returns
+    -------
+    vertices : tuple
+        The vertices of the texture square.
+    """
+    m = 1.0 / n
+    dx = x * m
+    dy = y * m
+    return dx, dy, dx + m, dy, dx + m, dy + m, dx, dy + m
+
+
+def tex_coords(top: tuple, bottom: tuple, side: tuple) -> list:
+    """Return a list of the texture squares for the top, bottom and side.
+
+    Parameters
+    ----------
+    top, bottom, side : tuple of len 2
+        The x, y coordinates of the top left corner of the texture square.
+
+    Returns
+    -------
+    tex_coords : list
+        The texture coordinates for the top, bottom and side.
+    """
+    top = tex_coord(*top)
+    bottom = tex_coord(*bottom)
+    side = tex_coord(*side)
+    result = []
+    result.extend(top)
+    result.extend(bottom)
+    result.extend(side * 4)
+    return result
+
+WINDOW_WIDTH = 800
+WINDOW_HEIGHT = 600
+TICKS_PER_SEC = 60
+SECTOR_SIZE = 16  # Size of sectors used to ease block loading.
+TEXTURE_PATH = 'assets/texture.png'
```

Comment:
Test return 3 errors

Reply:
Looking into it, it's because the pipeline isn't running the `Run poetry install --no-interaction --no-root` step, nor is it running the `Run sudo apt install freeglut3-dev` step. Those two steps not being run would cause _all_ the tests to fail since none of the dependencies were correctly installed.

If you look at the [first ci attempt ran](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/actions/runs/5419416487/attempts/1?pr=63), the PR passed all test. 

@xieWright Can you look into the CI job please?

Reply:
Looking into it a bit more its a caching problem on the CI job. I suspect the caches aren't being restored correctly.

Reply:
Found the issue, a fix has been merged. #64